src/amd64/amd64-51-30k-choose_tp.s

```diff
@@ -0,0 +1,2189 @@
+movq   200(%rcx,%rdi),%rbp
+
+# qhasm: txaddy0 = t if =
+# asm 1: cmove <t=int64#15,<txaddy0=int64#10
+# asm 2: cmove <t=%rbp,<txaddy0=%r12
+cmove %rbp,%r12
+
+# qhasm: t = *(uint64 *)(basep + 208 + pos)
+# asm 1: movq   208(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   208(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   208(%rcx,%rdi),%rbp
+
+# qhasm: txaddy1 = t if =
+# asm 1: cmove <t=int64#15,<txaddy1=int64#11
+# asm 2: cmove <t=%rbp,<txaddy1=%r13
+cmove %rbp,%r13
+
+# qhasm: t = *(uint64 *)(basep + 216 + pos)
+# asm 1: movq   216(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   216(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   216(%rcx,%rdi),%rbp
+
+# qhasm: txaddy2 = t if =
+# asm 1: cmove <t=int64#15,<txaddy2=int64#12
+# asm 2: cmove <t=%rbp,<txaddy2=%r14
+cmove %rbp,%r14
+
+# qhasm: t = *(uint64 *)(basep + 184 + pos)
+# asm 1: movq   184(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   184(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   224(%rcx,%rdi),%rbp
+
+# qhasm: txaddy3 = t if =
+# asm 1: cmove <t=int64#15,<txaddy3=int64#13
+# asm 2: cmove <t=%rbp,<txaddy3=%r15
+cmove %rbp,%r15
+
+# qhasm: t = *(uint64 *)(basep + 232 + pos)
+# asm 1: movq   232(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   232(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   232(%rcx,%rdi),%rbp
+
+# qhasm: txaddy4 = t if =
+# asm 1: cmove <t=int64#15,<txaddy4=int64#14
+# asm 2: cmove <t=%rbp,<txaddy4=%rbx
+cmove %rbp,%rbx
+
+# qhasm: =? u - 3
+# asm 1: cmp  $3,<u=int64#5
+# asm 2: cmp  $3,<u=%r8
+cmp  $3,%r8
+
+# qhasm: t = *(uint64 *)(basep + 320 + pos)
+# asm 1: movq   320(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   320(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   320(%rcx,%rdi),%rbp
+
+# qhasm: tysubx0 = t if =
+# asm 1: cmove <t=int64#15,<tysubx0=int64#2
+# asm 2: cmove <t=%rbp,<tysubx0=%rsi
+cmove %rbp,%rsi
+
+# qhasm: t = *(uint64 *)(basep + 328 + pos)
+# asm 1: movq   328(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   328(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   328(%rcx,%rdi),%rbp
+
+# qhasm: tysubx1 = t if =
+# asm 1: cmove <t=int64#15,<tysubx1=int64#6
+# asm 2: cmove <t=%rbp,<tysubx1=%r9
+cmove %rbp,%r9
+
+# qhasm: t = *(uint64 *)(basep + 336 + pos)
+# asm 1: movq   336(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   336(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   336(%rcx,%rdi),%rbp
+
+# qhasm: tysubx2 = t if =
+# asm 1: cmove <t=int64#15,<tysubx2=int64#7
+# asm 2: cmove <t=%rbp,<tysubx2=%rax
+cmove %rbp,%rax
+
+# qhasm: t = *(uint64 *)(basep + 344 + pos)
+# asm 1: movq   344(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   344(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   344(%rcx,%rdi),%rbp
+
+# qhasm: tysubx3 = t if =
+# asm 1: cmove <t=int64#15,<tysubx3=int64#8
+# asm 2: cmove <t=%rbp,<tysubx3=%r10
+cmove %rbp,%r10
+
+# qhasm: t = *(uint64 *)(basep + 272 + pos)
+# asm 1: movq   272(<basep=int64#4,<pos=int64#1),>t=int64#15
+# asm 2: movq   272(<basep=%rcx,<pos=%rdi),>t=%rbp
+movq   352(%rcx,%rdi),%rbp
```

Comment:
This one also has the "new" offset in comments. Given I have no idea what data this is trying to copy, I can't guess at what the bug is.

Reply:
And the same oddities continue later, I'm not pointing them out.